tests/components/braviatv/test_config_flow.py

```diff
@@ -0,0 +1,248 @@
+"""Define tests for the Bravia TV config flow."""
+from collections import OrderedDict
+
+from asynctest import patch
+
+from homeassistant import data_entry_flow
+from homeassistant.components.braviatv.const import CONF_IGNORED_SOURCES, DOMAIN
+from homeassistant.config_entries import SOURCE_IMPORT, SOURCE_USER
+from homeassistant.const import CONF_HOST, CONF_PIN
+
+from tests.common import MockConfigEntry
+
+BRAVIA_SYSTEM_INFO = {
+    "product": "TV",
+    "region": "XEU",
+    "language": "pol",
+    "model": "TV-Model",
+    "serial": "serial_number",
+    "macAddr": "AA:BB:CC:DD:EE:FF",
+    "name": "BRAVIA",
+    "generation": "5.2.0",
+    "area": "POL",
+    "cid": "very_unique_string",
+}
+
+BRAVIA_SOURCE_LIST = OrderedDict(
+    [
+        ("HDMI 1", "extInput:hdmi?port=1"),
+        ("HDMI 2", "extInput:hdmi?port=2"),
+        ("HDMI 3/ARC", "extInput:hdmi?port=3"),
+        ("HDMI 4", "extInput:hdmi?port=4"),
+        ("AV/Component", "extInput:component?port=1"),
+    ]
+)
+
+CONFIG = {CONF_HOST: "bravia-host", CONF_PIN: "1234"}
+
+config_entry = MockConfigEntry(
+    domain=DOMAIN,
+    unique_id="very_unique_string",
+    data={CONF_HOST: "bravia-host", CONF_PIN: "1234"},
+    title="TV-Model",
+)
+
+
+async def test_show_form(hass):
+    """Test that the form is served with no input."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": SOURCE_USER}
+    )
+
+    assert result["type"] == data_entry_flow.RESULT_TYPE_FORM
+    assert result["step_id"] == SOURCE_USER
+
+
+async def test_import(hass):
+    """Test that the import works."""
+    with patch("bravia_tv.BraviaRC.connect", return_value=True), patch(
```

Comment:
Please also patch `async_setup_entry` in the integration component module to avoid I/O in the test.

Eg:
https://github.com/home-assistant/core/blob/bc26be3c11ea40ed8d423f35a0fcbaf7b9cf7cb4/tests/components/hue/test_config_flow.py#L19-L23